incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -0,0 +1,201 @@
+		setupV1alpha1(hncFromVersion)
+	})
+
+	AfterEach(func() {
+		CleanupNamespaces(nsA, nsB)
+		// Only tear down HNC at HEAD, since that's what we just deployed.
+		TearDownHNC("")
+	})
+
+	It("should convert subnamespace anchors", func() {
+		// Before conversion, create namespace A and a subnamespace B.
+		MustRun("kubectl create ns", nsA)
+		anchor := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: SubnamespaceAnchor
+metadata:
+  name: e2e-conversion-test-b
+  namespace: e2e-conversion-test-a`
+		MustApplyYAML(anchor)
+		RunShouldContain(nsB, 10, "kubectl get ns")
+		RunShouldContain("ok", 10, "kubectl get subnamespaceanchors.hnc.x-k8s.io", nsB, "-n", nsA, "-oyaml")
+
+		// Convert
+		setupV1alpha2()
+
+		// Verify
+		verifyCRDConversion()
+		MustRun("kubectl -n", nsA, "delete subnamespaceanchors.hnc.x-k8s.io", nsB)
+		RunShouldNotContain(nsB, 4, "kubectl get ns")
+	})
+
+	It("should convert HCs with parent", func() {
+		// Before conversion, create namespace A and B and set A as the parent of B.
+		MustRun("kubectl create ns", nsA)
+		MustRun("kubectl create ns", nsB)
+		hc := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: HierarchyConfiguration
+metadata:
+  name: hierarchy
+  namespace: e2e-conversion-test-b
+spec:
+  parent: e2e-conversion-test-a`
+		MustApplyYAML(hc)
+
+		// Convert
+		setupV1alpha2()
+
+		// Verify CRD conversion and kubectl plugin with v1alpha2.
+		verifyCRDConversion()
+		RunShouldContain(nsB, 5, "kubectl hns tree", nsA)
+	})
+
+	It("should convert HCs with allowCascadingDelete", func() {
+		// Before conversion, create namespace A with allowCascadingDelete.
+		MustRun("kubectl create ns", nsA)
+		anchor := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: HierarchyConfiguration
+metadata:
+  name: hierarchy
+  namespace: e2e-conversion-test-a
+spec:
+  allowCascadingDelete: true`
+		MustApplyYAML(anchor)
+
+		// Convert
+		setupV1alpha2()
+
+		// Verify
+		verifyCRDConversion()
+		RunShouldContain("is already set to true; unchanged", 2, "kubectl hns set", nsA, "-a")
+	})
+
+	It("should still have HC condition if it exists in v1alpha1", func() {
+		// Before conversion, create namespace B with a missing parent A (have to
+		// create A first and then delete it because otherwise the webhook will deny
+		// setting a non-existing namespace as parent).
+		MustRun("kubectl create ns", nsA)
+		MustRun("kubectl create ns", nsB)
+		hc := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: HierarchyConfiguration
+metadata:
+  name: hierarchy
+  namespace: e2e-conversion-test-b
+spec:
+  parent: e2e-conversion-test-a`
+		MustApplyYAML(hc)
+		MustRun("kubectl delete ns", nsA)
+
+		// Convert
+		setupV1alpha2()
+
+		// Verify CRD conversion and the condition.
+		verifyCRDConversion()
+		RunShouldContain("CritParentMissing: missing parent", 2, "kubectl hns describe", nsB)
```

Comment:
Again, use `get -oyaml` so we can directly observe the v1alpha2 conditions when we have them.